src/EditProvider.ts

```diff
@@ -99,16 +100,84 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    path: string = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      /**
+       * Use a different `.unibeautify` config when vscode config is set and
+       * define if the project config should extend the vscode config.
+       *
+       * @argument {unibeautify.defaultConfig} string path to a file
+       * @argument {unibeautify.extendConfig} boolean extend or not
+       */
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+      let defaultConfigOverride = vscodeSettings.defaultConfig;
+      let extendConfig = vscodeSettings.extendConfig;
+      let workspace = path;
+
+      const explorerOptions: any = {};
+      if (defaultConfigOverride) {
+        path = defaultConfigOverride;
+      }
+
+      const explorer = cosmiconfig("unibeautify", explorerOptions);
       const defaultConfig: LanguageOptionValues = {};
-      return explorer
-        .search(filePath)
-        .then(result => (result ? result.config : defaultConfig));
+
+      let isDirectory = fs.lstatSync(path).isDirectory();
```

Comment:
@muuvmuuv I don't think I want `unibeautify.defaultConfig` to support being a path to a directory. I would prefer it only support file path to `.unibeautifyrc` file and use `explorer.load(filePath)`.

cc @szeck87 thoughts?

Reply:
I think both are good because you can choose your project root and it will search by its own for a config file. I‘ll exclude it in the next pr for now

Reply:
Searching from project root should definitely be supported, as it is now.

However, the `unibeautify.defaultConfig` option should be a file and not a directory, as this should only be used for non-standard and out-of-project-root configuration file.